xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -491,19 +492,23 @@ ManagedChannel createChannel(List<ServerInfo> servers) {
         ServerInfo serverInfo = servers.get(0);
         String serverUri = serverInfo.getServerUri();
         List<ChannelCreds> channelCredsList = serverInfo.getChannelCredentials();
-        ManagedChannel ch = null;
+        ManagedChannelBuilder<?> channelBuilder = null;
         // Use the first supported channel credentials configuration.
         // Currently, only "google_default" is supported.
         for (ChannelCreds creds : channelCredsList) {
           if (creds.getType().equals("google_default")) {
-            ch = GoogleDefaultChannelBuilder.forTarget(serverUri).build();
+            channelBuilder = GoogleDefaultChannelBuilder.forTarget(serverUri);
             break;
           }
         }
-        if (ch == null) {
-          ch = ManagedChannelBuilder.forTarget(serverUri).build();
+        if (channelBuilder == null) {
+          channelBuilder = ManagedChannelBuilder.forTarget(serverUri);
         }
-        return ch;
+
+        return channelBuilder
+            .keepAliveTime(5, TimeUnit.MINUTES)
+            .keepAliveTimeout(20, TimeUnit.SECONDS)
```

Comment:
DO we really want to hardcode these values? WHat will be ideal is if we can supply these values from the Bootstrap file as `XdsNameResolver.start` -> `XdsClientImpl` ctor -> `XdsClient.createChannel()` so these are configurable at runtime. The defaults could be these hardcoded values.

Reply:
While we could, most value is provided by having _some_ value here. Anything via bootstrap would be an optimization. It seems there is a wide range of "acceptable" values, so this will probably be "good enough" for everyone. We can maybe add a configuration option in the future, but there are _much_ more important things to do and we have limited resources.

Reply:
> Anything via bootstrap would be an optimization.

I think it is more than just optimization. Because of network conditions if someone wants to change these at runtime instead of having to recompile gRPC library they will be able to do that. But I agree this might be sufficient for now and there are more important things to do.

Reply:
It is "just an optimization" because xds LB should be able to function for 5 minutes without XDS server. That is to say it is not a correctness issue.